interview-questions.md

```diff
@@ -8,50 +8,52 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Your answer:
+Your answer: `map()` takes 3 parameters: `(value, array, index)`.  `value` is the required parameter, which represents the object in the current iteration.  `array` is optional. It refers to the array that we are iterating through. `index` is also optional, and refers to the index number of the element being used in the current iteration.
 
-Researched answer:
+Researched answer: I couldn't find anything additional for this question. There's more to `map()`, but not as it relates to this from what I can tell.
 
 3. What is the difference between map and filter?
 
-Your answer:
-
-Researched answer:
+Your answer: `map()` iterates through an array and returns a new array of equal size containing the transformed elements.  `filter()` iterates through the array and checks for the 'truthiness' of each element when passed to the block. `filter()` then puts all elements that evaluated to `true` into a new array and returns the filtered array.  This can be smaller than the original array. 
 
-1. What is iteration?
+Researched answer: Even if `map()` cannot properly evaluate an element, it will return `undefined` in its place. the array creaetd by `filter()` points to the same locations in memory as the original values. Altering this array could mutate the original.
 
-Your answer:
+4. What is iteration?
 
-Researched answer:
+Your answer: Iteration is performing an action repeatedly until a certain condition is met.  Examples are loops, `map()`, and `filter()`.  The conditions can be automatic (iterating until each element in the array has been run like `map()`), or manual (inserting a `break` command)
 
-1. What is the difference between a function and a method?
+Researched answer: Iteration according to the MDN docs is looping an action over an object until completion or a condition is met.
 
-Your answer:
+Your answer: A function is a reusable piece of self-contained, static code that can take in arguments (if needed) and provide a return.  A method is a type of function that runs inside an object.
 
-Researched answer:
+Researched answer: Methods are considered properties of objects that contain a function definition.  Basically methods are functions that are stored as properties.
 
-6. STRETCH: What is hoisting in JavaScript?
+5. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer:I *believe* hoisting is the ability to call a function before it has been defined in the code.  e.g. invoking a function on line 5 of your code, but having written the function on line 30.  JavaScript conceptually 'hoists' it up to the top to run the code.
 
-Researched answer:
+Researched answer: Hoisting is JavaScript moving all declarations to the top of the current scope.  This makes it a good rule to declare all variables at the top of their respective scopes.  Hoisting with `var` initializes the value to `undefined`. Hoisting `let` and `const` are not initialized and will result in an error.
```

Comment:
You are correct! Hoisting is generally bad practice, as it could lead to potential issues. This is part of the reason why in modern JavaScript, it is recommended to use 'let' and 'const' instead of 'var', and those two are block-scoped declarations while 'var' is a function-scoped declaration.

TLDR: declare your variables and functions before you use them haha

Reply:
Thanks for this feedback :)